addons/purchase_stock/models/stock_move.py

```diff
@@ -12,19 +12,19 @@ class StockMove(models.Model):
     purchase_line_id = fields.Many2one(
         'purchase.order.line', 'Purchase Order Line',
         ondelete='set null', index='btree_not_null', readonly=True)
-    created_purchase_line_id = fields.Many2one(
-        'purchase.order.line', 'Created Purchase Order Line',
-        ondelete='set null', index='btree_not_null', readonly=True, copy=False)
+    created_purchase_line_ids = fields.Many2many(
+        'purchase.order.line', 'stock_move_created_purchase_line',
+        'move_id', 'created_purchase_line_id', 'Created Purchase Order Lines', copy=False)
 
     @api.model
     def _prepare_merge_moves_distinct_fields(self):
         distinct_fields = super(StockMove, self)._prepare_merge_moves_distinct_fields()
-        distinct_fields += ['purchase_line_id', 'created_purchase_line_id']
+        distinct_fields += ['purchase_line_id', 'created_purchase_line_ids']
```

Comment:
You could remove it now, it's a many2many. Maybe you could merge the created_purchase_line_ids of different moves. To explore. Maybe another task